packages/assets-controllers/src/NftController.test.ts

```diff
@@ -3435,4 +3436,229 @@ describe('NftController', () => {
+
+      expect(spy).toHaveBeenCalledTimes(0);
+      expect(
+        nftController.state.allNfts[selectedAddress][SEPOLIA.chainId][0],
+      ).toStrictEqual({
+        address: '0xtest',
+        description: '',
+        favorite: false,
+        image: '',
+        isCurrentlyOwned: true,
+        name: '',
+        standard: 'ERC721',
+        tokenId: '3',
+      });
+    });
+
+    it('should  update metadata when some calls to fetch metadata succeed', async () => {
+      const { nftController } = setupController();
+      const { selectedAddress } = nftController.config;
+      const spy = jest.spyOn(nftController, 'updateNft');
+      const testNetworkClientId = 'sepolia';
+      // Add nfts
+      await nftController.addNft('0xtest1', '1', {
+        nftMetadata: {
+          name: '',
+          description: '',
+          image: '',
+          standard: 'ERC721',
+        },
+        networkClientId: testNetworkClientId,
+      });
+      await nftController.addNft('0xtest2', '2', {
+        nftMetadata: {
+          name: '',
+          description: '',
+          image: '',
+          standard: 'ERC721',
+        },
+        networkClientId: testNetworkClientId,
+      });
+      await nftController.addNft('0xtest3', '3', {
+        nftMetadata: {
+          name: '',
+          description: '',
+          image: '',
+          standard: 'ERC721',
+        },
+        networkClientId: testNetworkClientId,
+      });
+      sinon
+        // TODO: Replace `any` with type
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .stub(nftController, 'getNftInformation' as any)
+        .onFirstCall()
+        .returns({
+          name: 'name pudgy 1',
+          image: 'url pudgy 1',
+          description: 'description pudgy 2',
+        })
+        .onSecondCall()
+        .returns({
+          name: 'name pudgy 2',
+          image: 'url pudgy 2',
+          description: 'description pudgy 2',
+        })
+        .onThirdCall()
+        .rejects(new Error('Error'));
+
+      const testInputNfts: Nft[] = [
+        {
+          address: '0xtest1',
+          description: null,
+          favorite: false,
+          image: null,
+          isCurrentlyOwned: true,
+          name: null,
+          standard: 'ERC721',
+          tokenId: '1',
+        },
+        {
+          address: '0xtest2',
+          description: null,
+          favorite: false,
+          image: null,
+          isCurrentlyOwned: true,
+          name: null,
+          standard: 'ERC721',
+          tokenId: '2',
+        },
+        {
+          address: '0xtest3',
+          description: null,
+          favorite: false,
+          image: null,
+          isCurrentlyOwned: true,
+          name: null,
+          standard: 'ERC721',
+          tokenId: '3',
+        },
+      ];
+
+      await nftController.updateNftMetadata(testInputNfts, testNetworkClientId);
```

Comment:
```suggestion
      await nftController.updateNftMetadata({
        nfts: testInputNfts,
        networkClientId: testNetworkClientId,
      });
```

If you decide to go with the options object suggestion this change will be necessary.